api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3009,24 +3010,26 @@ public void removeObject(String bucketName, String objectName)
           private Result<DeleteError> error;
           private Iterator<DeleteError> errorIterator;
           private boolean completed = false;
-          private Iterator<String> objectNameIter = objectNames.iterator();
+          // divide the objectNames to sublists of the given batchSize
+          private Iterator<List<String>> subObjectListsIter =
+                  Iterables.partition(objectNames, 1000).iterator();
 
           private synchronized void populate() {
             List<DeleteError> errorList = null;
             try {
-              List<DeleteObject> objectList = new LinkedList<DeleteObject>();
-              int i = 0;
-              while (objectNameIter.hasNext() && i < 1000) {
-                objectList.add(new DeleteObject(objectNameIter.next()));
-                i++;
-              }
-
-              if (i > 0) {
-                errorList = removeObject(bucketName, objectList);
+              if (subObjectListsIter.hasNext()) {
+                List<DeleteObject> subDeleteObjectList = subObjectListsIter.next().stream().map(objectName -> {
+                  try {
+                    return new DeleteObject(objectName);
+                  } catch (XmlPullParserException e) {
+                    throw new RuntimeException(e);
+                  }
+                }).collect(Collectors.toList());
```

Comment:
I don't know what advantages we get here. My choice is `for` loop which is quite simple (and to read) than java8 stream here.  Do we get any performance improvement here?